src/test/ui/wf/wf-in-foreign-fn-decls-issue-80468.stderr

```diff
@@ -4,14 +4,22 @@ error[E0726]: implicit elided lifetime not allowed here
 LL | impl Trait for Ref {}
    |                ^^^- help: indicate the anonymous lifetime: `<'_>`
 
-error[E0308]: mismatched types
+error: incompatible lifetime on type
   --> $DIR/wf-in-foreign-fn-decls-issue-80468.rs:16:21
    |
 LL |     pub fn repro(_: Wrapper<Ref>);
-   |                     ^^^^^^^^^^^^ lifetime mismatch
+   |                     ^^^^^^^^^^^^
    |
-   = note: expected trait `Trait`
-              found trait `Trait`
+note: because of this predicate
+  --> $DIR/wf-in-foreign-fn-decls-issue-80468.rs:8:23
+   |
+LL | pub struct Wrapper<T: Trait>(T);
+   |                       ^^^^^
```

Comment:
In this case the `'static` lifetime obligation is introduced by the type parameter (`T: Trait` is implicitly `'sized`). It'd be nice to proactively handle this, detect this case and mention that.

Reply:
Actually, this is because the lifetime on `Ref` in `impl Trait for Ref {}` falls back to `'static` in resolution.

The lifetime for `Ref` in `pub fn repro(_: Wrapper<Ref>)` on the other hand defaults to an anonymous lifetimes.

A diagnostic here still might be useful to say something like "this lifetime is being assumed as `'static'`"

Reply:
I think an extra diagnostic here might make sense for another PR?